src/query/outputs.go

```diff
@@ -1,18 +1,45 @@
 package query
 
 import (
+	"encoding/json"
 	"fmt"
+	"os"
 	"path/filepath"
 
 	"github.com/thought-machine/please/src/core"
 )
 
 // TargetOutputs prints all output files for a set of targets.
-func TargetOutputs(graph *core.BuildGraph, labels []core.BuildLabel) {
+func TargetOutputs(graph *core.BuildGraph, labels []core.BuildLabel, useJSON bool) {
+	if useJSON {
+		targetOutputsJSON(graph, labels)
+	} else {
+		targetOutputsFlat(graph, labels)
+	}
+}
+
+func targetOutputsFlat(graph *core.BuildGraph, labels []core.BuildLabel) {
 	for _, label := range labels {
 		target := graph.TargetOrDie(label)
 		for _, out := range target.Outputs() {
 			fmt.Printf("%s\n", filepath.Join(target.OutDir(), out))
 		}
 	}
 }
+
+func targetOutputsJSON(graph *core.BuildGraph, labels []core.BuildLabel) {
+	data := map[string][]string{}
+	for _, label := range labels {
+		target := graph.TargetOrDie(label)
+		for _, out := range target.Outputs() {
+			data[label.String()] = append(data[label.String()], filepath.Join(target.OutDir(), out))
+		}
+	}
+	bs, err := json.Marshal(data)
```

Comment:
I think this should probably use MarshalIndent so it's a bit more readable (or at least doesn't have million-character-long lines).

Or, better, an Encoder with an indent set so it can stream the writes to stdout.